frame/staking/src/offchain_election.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Helpers for offchain worker election.
+
+use crate::{Call, CompactAssignments, Module, SessionInterface, Trait};
+use codec::Encode;
+use frame_system::offchain::{SubmitUnsignedTransaction};
+use sp_phragmen::{reduce, ExtendedBalance, PhragmenResult, StakedAssignment};
+use sp_std::{prelude::*, cmp::Ordering};
+use sp_runtime::{RuntimeAppPublic, RuntimeDebug};
+
+#[derive(RuntimeDebug)]
+pub(crate) enum OffchainElectionError {
+	/// No signing key has been found on the current node that maps to a validators. This node
+	/// should not run the offchain election code.
+	NoSigningKey,
+	/// Phragmen election returned None. This means less candidate that minimum number of needed
+	/// validators were present. The chain is in trouble and not much that we can do about it.
+	FailedElection,
+}
+
+/// Compares two sets of phragmen scores based on desirability and returns true if `that` is
+/// better `this`.
+///
+/// Evaluation is done in a lexicographic manner.
+pub(crate) fn is_score_better(this: [ExtendedBalance; 3], that: [ExtendedBalance; 3]) -> bool {
+	match that
+		.iter()
+		.enumerate()
+		.map(|(i, e)| e.cmp(&this[i]))
+		.collect::<Vec<Ordering>>()
+		.as_slice()
+	{
+		[Ordering::Greater, _, _] => true,
+		[Ordering::Equal, Ordering::Greater, _] => true,
+		[Ordering::Equal, Ordering::Equal, Ordering::Less] => true,
```

Comment:
```suggestion
		[Ordering::Equal, Ordering::Equal, Ordering::Greater] => true,
```
?

Reply:
The third parameter should be _minimised_.
https://github.com/paritytech/substrate/blob/2623d9883f09b083756e695a7b731b6cbf4d31e6/primitives/phragmen/src/lib.rs#L460